starter-code/src/clue.js

```diff
@@ -1,14 +1,165 @@
+        lastName: 'Plum',
+        occupation: 'Designer',
+        age: 22,
+        description: 'Billionaire video game designer',
+        image: 'https://66.media.tumblr.com/ee7155882178f73b3781603f0908617c/tumblr_phhxc7EhPJ1w5fh03_540.jpg', 
+        color: 'purple',
+    },
+
+    {
+        firstName: 'Kasandra',
+        lastName: 'Scarlet',
+        occupation: 'Actor',
+        age: 31,
+        description: 'She is an A-list movie star with a dark past',
+        image: 'https://www.radiotimes.com/uploads/images/Original/111967.jpg',
+        color: 'red',
+    },
+
+    {
+        firstName: 'Eleanor',
+        lastName: 'Peacock',
+        occupation: 'Socialité',
+        age: 36,
+        description: 'She is from a wealthy family and uses her status and money to earn popularity',
+        image: 'https://metrouk2.files.wordpress.com/2016/07/mrs-peacock.jpg',
+        color: 'blue',
+    },
+
+    {
+        firstName: 'Jack',
+        lastName: 'Mustard',
+        occupation: 'Retired Football player',
+        age: 62,
+        description: 'He is a former football player who tries to get by on his former glory',
+        image: 'https://static.independent.co.uk/s3fs-public/thumbnails/image/2016/07/04/08/unspecified-3.jpg',
+        color: 'yellow',
+    },
+];
 
 // Rooms Collection
-const roomsArray = [];
+const roomsArray = ['Dining Room',
+    'Conservatory',
+    'Kitchen',
+    'Study',
+    'Library',
+    'Billiard Room',
+    'Lounge',
+    'Ballroom',
+    'Hall',
+    'Spa',
+    'Living Room',
+    'Observatory',
+    'Theater',
+    'Guest House',
+    'Patio'];
 
 // Weapons Collection
-const weaponsArray = [];
+const weaponsArray = [
+    {
+        name: 'rope',  
+        weight: 10, 
+    },
+
+    {
+        name: 'knife', 
+        weight: 8,
+    },
+
+    {
+        name: 'candlestick', 
+        weight: 2,
+    },
+
+    {
+        name: 'dumbbell', 
+        weight: 30,
+    },
+
+    {
+        name: 'poison', 
+        weight: 2,
+    },
+
+    {
+        name: 'axe',
+        weight: 15,
+    },
+
+    {
+        name: 'bat', 
+        weight: 13,
+    },
+
+    {
+        name: 'trophy', 
+        weight: 25,
+    },
+
+    {
+        name: 'pistol', 
+        weight: 20
+    },
+];
 
 // ITERATION 2
 
+//Random function
+function selectRandom (arr) {
+    return arr[Math.floor(Math.random()*arr.length)];
+  }
+  
+  // console.log (selectRandom(suspectsArray));
+  // console.log (selectRandom(weaponsArray));
+  // console.log (selectRandom(roomsArray));
+
+function pickMystery () {
+    return {
+        suspect: selectRandom(suspectsArray),
+        room: selectRandom(roomsArray),
+        weapon: selectRandom(weaponsArray),
+    }
+}
+
 // ITERATION 3
+
+var mysteryEnvelope = pickMystery();
+
+function revealMystery (envelope) {
+  return `${envelope.suspect.firstName} ${envelope.suspect.lastName} killed Mr. Boddy using the ${envelope.weapon.name} in the ${envelope.room.name}!`
+};
+
```

Comment:
Here 'envelope.room.name' returns undefined because in the roomsArray, you didn't use any key 'name' to store the values of each room. So you would just need to write ${envelope.room} ;)